src/config/passport.js

```diff
@@ -10,4 +12,39 @@ passport.deserializeUser(async (id, done) => {
   done(null, USER);
 });
 
-// passport.use();
+passport.use(
+  new GoogleStrategy(
+    {
+      clientID: process.env.GAPP_CLIENT_ID,
+      clientSecret: process.env.GAPP_CLIENT_SECRET,
+      callbackURL: 'http://localhost:3000/api/auth/google/callback',
+    },
+    async (accessToken, refreshToken, profile, cb) => {
+      try {
+        let user = await User.findOne({ providerId: `google-${profile.id}` });
+        if (!user) {
+          user = await User.create({
+            email: profile.emails[0].value,
+            username:
+              profile.name.familyName +
+              profile.name.givenName +
+              '_' +
+              Math.floor(Math.random() * 89998 + 10000),
```

Comment:
As for creating a random number lets use the way it was shown here: https://stackoverflow.com/questions/8012002/create-a-unique-number-with-javascript-time